Dockerfile

```diff
@@ -36,18 +44,19 @@ USER composer
 # Run Composer Install
 RUN /usr/local/bin/composer install
 
+# Since .env is in .dockerignore, we have to get it from somewhere
+RUN cp .env.example .env
```

Comment:
Ahh, just realized it was the `php artisan key:generate` command that was failing initially and caused me to add the `cp` command. I wasn't seeing the error because I commented that line out. Now that I've uncommented it, I'm getting the exception again.

It looks like `php artisan key:generate` is explicitly requiring a `.env` file inside the container and fails if it doesn't find it, which breaks the docker build.

```
Step 16/18 : RUN php artisan key:generate
 ---> Running in f5e2c63c6c71

   ErrorException  : file_get_contents(/var/www/.env): failed to open stream: No such file or directory

  at /var/www/vendor/laravel/framework/src/Illuminate/Foundation/Console/KeyGenerateCommand.php:96
     92|     {
     93|         file_put_contents($this->laravel->environmentFilePath(), preg_replace(
     94|             $this->keyReplacementPattern(),
     95|             'APP_KEY='.$key,
  >  96|             file_get_contents($this->laravel->environmentFilePath())
     97|         ));
     98|     }
     99|
    100|     /**

  Exception trace:

  1   file_get_contents("/var/www/.env")
      /var/www/vendor/laravel/framework/src/Illuminate/Foundation/Console/KeyGenerateCommand.php:96

  2   Illuminate\Foundation\Console\KeyGenerateCommand::writeNewEnvironmentFileWith("base64:HN0NuNLB/+dL64ZPEjA4Y9c1aUSeYGGl9nGAg+7P2IY=")
      /var/www/vendor/laravel/framework/src/Illuminate/Foundation/Console/KeyGenerateCommand.php:80

  Please use the argument -v to see more details.
ERROR: Service 'app' failed to build: The command '/bin/sh -c php artisan key:generate' returned a non-zero code: 1
```

@Agares I'm open to suggestions if there's a better way to get around this, but simply removing it will un-fix the broken build issue that started me down this path in the first place.

My thinking is because this is a dev environment we could just remove `.env` from `.dockerignore`. Is it there for any particular reason?

Reply:
They `key:generate` must happen after the `.env` is generated. I don't think the `.env` should be included in the image, do you?

Reply:
Good point. I was kind of thinking that, but I figured since it's dev, maybe nobody cares.

If we want to remove the `.env` from the image, we could remove the `artisan` commands that rely on this file from the docker build and add them in an entrypoint script along with adding/generating the `.env` in the already running container.

Except we would still need a way to get around the fact that `.env` is in the `.dockerignore`. It doesn't get mounted, and I don't know if we can just copy it directly.